### PR TITLE
Adds Node Ignore from Deletion annotation

### LINKED ISSUE #184: Ability to Ignore Marked Nodes without Cordon
This issue is to add a feature to escalator to ignore nodes with a custom (Condition or annotation) from scale down events. The effect would be nodes can remain uncondoned and attached, but escalator won't delete it. These could also be exempt from tainting.

Something like `Condition`: `EscalatorIgnore`

Similar to a soft maintenance mode, where cordon isn't appropriate because the node still should get workloads.

### PR DESCRIPTION
**Pull request by Jacobious52**
Closes #184 

Adds the ability to mark a node safe from deletion with an annotation.
Annotation key `com.atlassian/escalator-ignore`
Value can be any non empty string. Can be used for reason of ignore.

Adds unit tests for `TryRemoveTaintedNodes` function

### CODE REVIEW

File: pkg/controller/scale_down.go [hunk 1]
```diff
@@ -12,6 +12,12 @@ import (
 	v1 "k8s.io/api/core/v1"
 )
 
+const (
+	// NodeEscalatorIgnoreAnnotation is the key of an annotation on a node that signifies it should be ignored from ASG deletion
+	// value does not matter, can be used for reason, as long as not empty
+	NodeEscalatorIgnoreAnnotation = "com.atlassian/escalator-ignore"
```
Comment (awprice) [inline pkg/controller/scale_down.go:18]: Why `com.atlassian`? We've used `atlassian.com` in the past in this repo. 

I think we should also signify in the annotation and the comment that the annotation only works for deletion, unlike cordoning. One might think this factors a node out of capacity calculations.

Comment (Jacobious52) [reply pkg/controller/scale_down.go]: changed to `atlassian.com/no-delete` and added comment

**Comment by Jacobious52** [comment]
Updated docs

File: docs/node-termination.md [hunk 2]
```diff
@@ -15,4 +15,40 @@ where we compare the creation timestamps of each node.
 
 This method is useful to ensure there are always new nodes in the cluster. If you want to deploy a configuration change
 to your nodes, you can use Escalator to cycle the nodes by terminating the oldest first until all of the nodes are
-using the latest configuration.
\ No newline at end of file
+using the latest configuration.
+
+### Annotating nodes / Stopping termination of selected nodes
+
+For most cases when wanting to exclude a node from termination for maintenance, you should first consider if the Cordon function [**Cordoning of nodes**](./scale-process).
```
Comment (awprice) [inline docs/node-termination.md:22]: `you should first consider the [**cordon function**](./scale-process).`

Comment (Jacobious52) [reply docs/node-termination.md]: Updated

**Pull request merged**